raiden/tests/unit/test_node_queue.py

```diff
@@ -43,9 +43,190 @@ def test_delivered_message_must_clean_unordered_messages(chain_id):
     )
 
     chain_state.queueids_to_queues[queue_identifier] = [first_message, second_message]
-    delivered_message = state_change.ReceiveDelivered(message_identifier)
+
+    delivered_message = state_change.ReceiveDelivered(recipient, message_identifier)
+
+    iteration = node.handle_delivered(chain_state, delivered_message)
+    new_queue = iteration.new_state.queueids_to_queues.get(queue_identifier, [])
+
+    assert first_message not in new_queue
+
+    # Register 2 messages into the queue
+    chain_state.queueids_to_queues[queue_identifier] = [first_message, second_message]
+
+    delivered_message = state_change.ReceiveDelivered(recipient, message_identifier)
 
     iteration = node.handle_delivered(chain_state, delivered_message)
     new_queue = iteration.new_state.queueids_to_queues.get(queue_identifier, [])
 
     assert first_message not in new_queue
+
+
+def test_delivered_message(chain_id):
```

Comment:
These tests can be way more readable if the logic is factored out from `handle_delivered` and `handle_processed` and called directly, e.g.:

```python
queues = dict()
queues[queue_identifier] = [first_message, second_message]
node.inplace_remove_delivered(queues, Delivered(...))
assert ...
```